pkg/coscheduling/coscheduling.go

```diff
@@ -137,6 +138,25 @@ func (cs *Coscheduling) PreFilter(ctx context.Context, state *framework.CycleSta
 	return framework.NewStatus(framework.Success, "")
 }
 
+// PostFilter is use for rejecting the group pods if a pod does not pass PreFilter or Filter.
+func (cs *Coscheduling) PostFilter(ctx context.Context, state *framework.CycleState, pod *v1.Pod,
+	filteredNodeStatusMap framework.NodeToStatusMap) (*framework.PostFilterResult, *framework.Status) {
+	pgName, pg := cs.pgMgr.GetPodGroup(pod)
+	if pg == nil {
+		klog.V(4).Info("Pod does not belong to any group")
+		return &framework.PostFilterResult{}, framework.NewStatus(framework.Success, "")
+	}
+	cs.frameworkHandler.IterateOverWaitingPods(func(waitingPod framework.WaitingPod) {
```

Comment:
Now the logic is running unconditionally. Let's look at the possibles cases:

- current pod failed due to not meeting PG's `minAvailable`. In this case, there are no waiting pods, so the loop here is a no-op. Although may have some perf overhead, generally it's fine.
- current pod meets PG's `minAvailable` but failed due to other constraints. So in this case, IMO it's too aggressive to reject the other waiting Pods? Because per my understanding, this Pod may NOT be the bottleneck to block the whole PG? i.e., maybe another Pod later can be scheduled successfully and hence "approve()" the waiting pods. I'm open to discussion on this. @mateuszlitwin @denkensk @yuanchen8911 it'd be appreciable if you can share your thoughts.

Reply:
@Huang-Wei From my understanding, this pod does not block the whole PG. But this PG may block following PGs.

Reply:
@cwdsuzhou The current logic elimiated the first case I mentioned above. But the second case can still happen IMO. For example, the `minAvailable` is 3, however, there comes 5 Pods, so it doesn't sound like a good idea to reject all when the 3rd Pod failed the scheduling?

Reply:
@xq2005 can you elaborate? are you implying we still should reject all Pods?

Reply:
@Huang-Wei Case 2 example:  pg1 and pg2 are waiting for scheduling, gp1 minAvailable is 4 and pg2 minAvailable is 2.  At this time there are 3 free pods resource in the cluster. pg1's 3 pods are waiting and pg2's pods are "Unschedulable". 

I think sometime this feature is useful, but it is **too aggressive** and may import resource starvation. We need more discussion about it.

Reply:
@xq2005 We mean we should not reject all when we have 5 pods, I need minMember 3 pods, cluster resource can meet 3 but can not meet 5. At this time, we should not reject the 5 pods.

The case you meant would not happen
1. we will check resource 
2. If any one failed of 4 pods failed when filter. The whole group would be rejected and resource will be released.

Reply:
@cwdsuzhou I used resource "pods" as an example, other resources may trigger the same behavior. My case wants to show new logic is useful.

Refer Wei's comment
> For example, the minAvailable is 3, however, there comes 5 Pods, so it doesn't sound like a good idea to reject all when the 3rd Pod failed the scheduling?

The 4th/5th pod may successful. It is too aggressive to reject pg when meeting the first failed pod in this pg.